code/modules/mob/dead/observer/observer.dm

```diff
@@ -142,29 +142,22 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 
 
 /mob/dead/observer/Topic(href, href_list)
-	. = ..()
-	if(.)
-		return
-	switch(href_list[href])
-		if("reentercorpse")
```

Comment:
Why remove switch

Reply:
switch smells and if somehow there were more than one of them